### PR TITLE
further guarding of options for #932

### PR DESCRIPTION
**Pull request by davidcarlisle**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [X] Test file(s) added
- [x] Version and date string updated in changed source files
- [X] Relevant `\changes` entries in source included
- [X] Relevant `changes.txt` updated
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

I added rollback in ltdefns but changes in ltclass treated as bug fix on recent changes so covered by existing rollbacks undoing the raw list handling

### CODE REVIEW

File: base/ltclass.dtx [hunk 1]
```diff
@@ -1520,6 +1521,7 @@
    \ifx\@classoptionslist\relax\else
     \@for\CurrentOption:=\@classoptionslist\do{%
       \ifx\CurrentOption\@empty\else
+        \edef\CurrentOption{\detokenize\expandafter{\CurrentOption}}%
```
Comment (josephwright) [inline base/ltclass.dtx:1524]: Using `\detokenize` here in re-defining `\CurrentOption` means that this 'shows up' in `\@use@ption`: if we imagine for example an option called `foo{=}bar`, the detokenization would now strip the braces of meaning, and we'd get 'unused option `foo`' not 'unused option `foo{=}bar`'. I can see that is somewhat contrived, but want to be sure that the effect is deliberate/expected.

File: base/ltclass.dtx [hunk 2]
```diff
@@ -1557,13 +1559,15 @@
...
 % The common part of |\ProcessOptions| and |\ProcessOptions*|.
+% \changes{v1.5e}{2022/10/22}{Use \cs{detokenize}}
 %    \begin{macrocode}
 %</2ekernel>
 %<*2ekernel|latexrelease>
 %<latexrelease>\IncludeInRelease{2020/10/01}%
 %<latexrelease>                 {\@process@pti@ns}{Unused options issue}%
 \def\@process@pti@ns{%
   \@for\CurrentOption:=\@curroptions\do{%
+   \edef\CurrentOption{\detokenize\expandafter{\CurrentOption}}%
```
Comment (josephwright) [inline base/ltclass.dtx:1570]: Same point about `\detokenize`.

**Review by josephwright — APPROVED** [review]
I'm happy for this to be merged, just want to be sure the `\detokenize` effect is by-design

File: base/ltclass.dtx (hunk 1 above)
Comment (davidcarlisle) [reply base/ltclass.dtx:1524]: @josephwright  yes but this is the classic option handler and as existing behaviour is the low level csname error, there don't seem to be any compatibility issues, in practice only simple names are going to be declared, the only issue is to make sure foo=\whatever survives eough to get on the global class options list (without being detokenized)

**Comment by davidcarlisle** [comment]
@josephwright  cyrillic tlb2888 fails due to

```
    \expandafter\in@\expandafter{\CurrentOption}%
                                {T2A,T2B,T2C,X2,LCY,OT2}%
```

now  \CurrentOption is detokenized, we could live that or replace

`\edef\CurrentOption{\detokenize\expandafter{\CurrentOption}}`

by detokenizing locally each time it is used in a csname

**Comment by josephwright** [comment]
I suspect this might come up in other places: people expect `\CurrentOption` to be tokenized. So I think `\detokenize` will beed to be used on a need-to-apply basis.

File: base/ltclass.dtx (hunk 1 above)
Comment (davidcarlisle) [reply base/ltclass.dtx]: adjusted

**Pull request merged**